packages/frames.js/src/core/types.ts

```diff
@@ -196,6 +196,30 @@ export type FramesOptions<
    * @defaultValue '/'
    */
   basePath?: string;
+  /**
+   * Overrides the detected URL of the request. URL is used in combination with `basePath` to generate target URLs for Buttons.
+   * Provided value must be full URL with protocol and domain.
+   * This is useful if the URL detection fails to recognize the correct URL or if you want to override it.
+   *
+   * This URL also overrides the request.url value with the provided value.
+   *
+   * @example
+   * ```ts
+   * // using string, the value of ctx.url and request.url will be set to this value
+   * {
+   *  baseURL: 'https://example.com',
+   * }
+   * ```
+   *
+   * @example
+   * ```ts
+   * // using URL, the value of ctx.url and request.url will be set to this value
+   * {
+   * baseURL: new URL('https://example.com'),
+   * }
+   * ```
+   */
+  baseURL?: string | URL;
```

Comment:
we're using `httpUrl`, `postUrl`, `getTokenUrl` in other places, so should probably be `baseUrl` for consistency